### DISCUSSION TITLE
Unsure on the declaration/type hint of non mapped instance variables

### DISCUSSION THREAD

**Discussion by Goryaa** [sqlalchemy/sqlalchemy#10117] [Usage Questions] [answered]
I have a few cases where I have instance variable/s defined for a model that are unrelated to any columns/relationships/etc (Initialized through `__init__` and/or `@reconstructor`). In these cases we would type hint these like:
```
class Model(Base):
    __tablename__ = "parent"

    id: Mapped[int] = mapped_column(primary_key=True)

    _some_attr: int
```
With 2.0, `_some_attr: int` now throws an error:
```
sqlalchemy.exc.ArgumentError: Type annotation for "Model._some_attr" can't be co
rrectly interpreted for Annotated Declarative Table form.  ORM annotations shoul
d normally make use of the ``Mapped[]`` generic type, or other ORM-compatible ge
neric type, as a container for the actual type, which indicates the intent that
the attribute is mapped. Class variables that are not intended to be mapped by t
he ORM should use ClassVar[].  To allow Annotated Declarative to disregard legac
y annotations which don't use Mapped[] to pass, set "__allow_unmapped__ = True"
on the class or a superclass this class.
```
Is this way of defining a instance variable no longer supported due to the new `Mapped[...]` type hint?

Additionally, I also noticed a slight inconsistency:
 - `_some_attr: int` raises the error
 - `_some_attr: int = None` also raises the error
 - `_some_attr: int = 0` does not raise any error and works, as long as the value is not `None` or something like `mapped_column(...)`

**Comment by zzzeek** [comment, marked as answer]
the purpose of `__allow_unmapped__ = True` is to allow these non-`Mapped` variables to be skipped.   Originally it was so that the 1.4 style of annotations with the "mypy" extension would be allowed.   We dont default to this setting because we feel it would be an extremely common error (one I make myself a lot) to forget the `Mapped` part.

if this is a matter of the error message being a little too heavy handed with the term "legacy", we can adjust that.
